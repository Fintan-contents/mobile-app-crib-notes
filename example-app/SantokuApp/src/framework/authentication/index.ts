export {AuthenticationService, ActiveAccountIdNotFoundError, PasswordNotFoundError} from './AuthenticationService';
export * from './SecureStorageAdapter';
export * from './UnauthorizedError';
