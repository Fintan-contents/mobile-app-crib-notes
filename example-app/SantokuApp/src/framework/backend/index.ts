export * from './refreshCsrfToken';
export * from './refreshSession';
