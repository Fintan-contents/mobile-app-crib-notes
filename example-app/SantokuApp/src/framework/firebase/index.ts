export {firebaseConfig, FirebaseConfig} from './FirebaseConfig';
