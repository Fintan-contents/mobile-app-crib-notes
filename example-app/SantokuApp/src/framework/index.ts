export {AuthenticationService, ActiveAccountIdNotFoundError, PasswordNotFoundError} from './authentication';
export {AppConfig} from './config';
export {firebaseConfig, FirebaseConfig} from './firebase';
export type {Logger, LoggerOptions} from './logging';
export {log, createLogger} from './logging';
export {m, loadMessages, BundledMessagesLoader} from './message';
export {LocalAuthentication} from './local-authentication';
export {launchedId, generatePassword, useIsMounted} from './utilities';
