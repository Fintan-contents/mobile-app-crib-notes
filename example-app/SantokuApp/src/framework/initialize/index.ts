export * from './withInitialData';
export * from './useAppInitializer';
