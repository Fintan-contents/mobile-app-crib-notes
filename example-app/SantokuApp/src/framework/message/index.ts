export * from './Message';
export * from './BundledMessageLoader';
