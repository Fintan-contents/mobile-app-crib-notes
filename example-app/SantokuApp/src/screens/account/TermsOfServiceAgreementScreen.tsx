import * as SplashScreen from 'expo-splash-screen';
import React, {useEffect} from 'react';
import {StyleSheet, View} from 'react-native';
import {Button, Text} from 'react-native-elements';
import {useNavigateToAuthenticatedStackScreen} from 'screens/useNavigateToScreen';

const ScreenName = 'TermsOfServiceAgreement';
const Screen: React.FC = () => {
  const onGoToHomeScreen = useNavigateToAuthenticatedStackScreen('Home');

  // TODO: 本来は起動直後の初期化処理完了タイミングに持っていくべきなので、初期化処理実装時に併せて対応する
  useEffect(() => {
    // スプラッシュスクリーンを閉じる前に100ms程度は待たないと、コンポーネントのレンダリングが完了せず画面が一瞬白くなってしまいます。
    // expo-app-loadingでも200ms待っているので、ここでも200ms待つようにしています。
    // cf: https://github.com/expo/expo/blob/sdk-41/packages/expo-app-loading/src/AppLoadingNativeWrapper.tsx#L27
    setTimeout(() => {
      SplashScreen.hideAsync().catch(() => {
        /* Ignore error */
      });
    }, 200);
  }, []);

  return (
    <View style={styles.container} testID="TermsOfServiceAgreementScreen">
      <Text h1>利用規約</Text>
      <Button title="Go to Home" testID="goToHomeButton" onPress={onGoToHomeScreen} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: 'white',
  },
});

export const TermsOfServiceAgreementScreen = {
  component: Screen,
  name: ScreenName,
  option: {
    title: '利用規約',
    presentation: 'formSheet' as const,
  },
};
