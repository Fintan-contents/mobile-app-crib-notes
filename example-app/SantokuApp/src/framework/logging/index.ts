import {firebaseConfig} from '../firebase';
import {FirebaseCrashlyticsTransport} from './FirebaseCrashlyticsTransport';
import {createLogger, Logger, LoggerOptions} from './Logger';

const log =
  __DEV__ || firebaseConfig.isDummy
    ? createLogger({level: 'trace'})
    : createLogger({level: 'error', transports: [new FirebaseCrashlyticsTransport()]});

export type {Logger, LoggerOptions};
export {log, createLogger};
