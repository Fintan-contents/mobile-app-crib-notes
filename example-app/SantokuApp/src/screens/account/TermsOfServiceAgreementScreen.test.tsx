import '@testing-library/jest-native/extend-expect';
import {render} from '@testing-library/react-native';
import {BundledMessagesLoader, loadMessages} from 'framework';
import React from 'react';

import {TermsOfServiceAgreementScreen} from './TermsOfServiceAgreementScreen';

beforeAll(async () => {
  await loadMessages(new BundledMessagesLoader());
});

describe('TermsOfServiceAgreement', () => {
  it('マウントされたときに正常にレンダリングされること', () => {
    const app = render(<TermsOfServiceAgreementScreen.component />);
    expect(app.queryByTestId('TermsOfServiceAgreementScreen')).not.toBeNull();
    expect(app).toMatchSnapshot();
  });
});
