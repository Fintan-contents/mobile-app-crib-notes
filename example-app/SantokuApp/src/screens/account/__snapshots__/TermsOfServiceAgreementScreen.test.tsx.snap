// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TermsOfServiceAgreement スプラッシュスクリーンを閉じるのに失敗しても何もしないこと 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="TermsOfServiceAgreementScreen"
>
  <Text
    style={
      Object {
        "color": "#242424",
        "fontSize": 51.42857142857143,
      }
    }
  >
    利用規約
  </Text>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        undefined,
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="goToHomeButton"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2089dc",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Go to Home
        </Text>
      </View>
    </View>
  </View>
</View>
`;
